Remove some more cranelift-filetest wasm testing bits

alexcrichton opened the pull request:
Some more cleanup enabled through #8140 and #8129

cfallin commented on tests/disas.rs:
```diff
@@ -227,3 +247,117 @@ impl Test {
+        return Ok(());
+    }
+
+    if std::env::var("CRANELIFT_TEST_BLESS").unwrap_or_default() == "1" {
```
Idle thought: if the test is now a Wasmtime test, should this environment variable be something like `WASMTIME_TEST_BLESS`?

(Smallest of small bikeshed color comments, fine either way, just a thought...)

bjorn3 commented on cranelift/filetests/src/test_wasm.rs:
```diff
@@ -1,198 +0,0 @@
-
-    let wasm = wat::parse_str(wat).context("failed to parse the test WAT")?;
-    let mut validator = wasmparser::Validator::new_with_features(
-        cranelift_wasm::ModuleEnvironment::wasm_features(&env),
```
Can the cranelift-wasm and wasmparser dependencies be removed from filetests?

alexcrichton replied:
Oh indeed they can!

alexcrichton commented:
This [hit an unusual error on CI](https://github.com/bytecodealliance/wasmtime/actions/runs/8298501522/job/22712023968) which looked like:

```
     Running tests/filetests.rs (target/riscv64gc-unknown-linux-gnu/debug/deps/filetests-9d21d101a4f160f6)
thread 'worker #2' panicked at cranelift/jit/src/backend.rs:342:40:
can't resolve libcall ceilf
```

Apparently the removal of the wasm-related dependencies also removed all dependencies on libm for riscv64 which meant that tests were failing as a dlsym for various libcall symbols was failing. I just added a commit which forces linking to libm by referring to a symbol and calling it.
